src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -1,109 +1,165 @@
+                $diffArray[$versionKey]['changes'] = $changes;
+            } else {
+                unset($diffArray[$versionKey]);
             }
-            $diffArray[$versionKey] = $this->getCurrentVersionElement($currentVersion, $diffArray[$versionKey]);
         }
 
         return $this->removeColumnIfOnlyUnchanged($diffArray);
     }
 
-    /**
-     * Method for getting associations for given entity.
-     * @param Object $entity object with entity
-     * @return array entities
-     */
-    protected function getAssociations($entity)
+    public function getVersionsOfProperty($entity, $columnName)
     {
-        $associations = $this->em->getRepository(Association::class)->findBy(array('class' => get_class($entity), 'fk' => $entity->getId()));
-        return $associations;
+        // doc is in interface
+
+        $allVer = $this->getAllVersionsDiff($entity);
+        /*
+         * Could filter in db, but some properties are in diff, some are in an association.
+         * And only for associations: $columnName = $this->getColumnNameForAssociation($propertyName);
+         */
+
+        return $this->filterLogForOneProperty($allVer, $columnName);
     }
 
     /**
      * Method creating version key (to group table entries associated with the same user action).
-     * @param \DataDog\AuditBundle\Entity\AuditLog $currentVersion entity for which version key is created
-     * @param array $diffArray subsequent elements are diff for each version
+     *
+     * @param AuditLog $currentVersion entity for which version key is created
+     * @param array    $diffArray      subsequent elements are diff for each version
+     *
      * @return string version key (timestamp and user id)
      */
-    protected function getVersionKey(AuditLog $currentVersion, $diffArray)
+    protected function getVersionKey(AuditLog $currentVersion, array $diffArray)
     {
         $versionTimestamp = $currentVersion->getLoggedAt()->getTimestamp();
-        $versionTimestampBefore = $currentVersion->getLoggedAt()->getTimestamp() - 1; // 1 second before by the same user
-        $versionTimestampAfter = $currentVersion->getLoggedAt()->getTimestamp() + 1; // 1 second after by the same user
 
-        $versionUser = $currentVersion->getBlame()->getFk();
+        $versionUser = $currentVersion->getBlame() ? $currentVersion->getBlame()->getFk() : -1;
         $versionKeyNormal = sprintf('%d_%d', $versionTimestamp, $versionUser); // having user and time prevent from logging in the same place simoultaneus changes from more then one user
-        $versionKeyBefore = sprintf('%d_%d', $versionTimestampBefore, $versionUser);
-        $versionKeyAfter = sprintf('%d_%d', $versionTimestampAfter, $versionUser);
+        $versionKeyBefore = sprintf('%d_%d', $versionTimestamp - 1, $versionUser); // 1 second before by the same user
+        $versionKeyAfter = sprintf('%d_%d', $versionTimestamp + 1, $versionUser); // 1 second after by the same user
```

Comment:
I really do not understand the idea of this change. Both approaches are for me generally correct, but I think code is more clear if at the beginning we declare variables and and then real logic happen. What's more, that code would be more portable because you can just copy&paste it to new function if our future development needs that (too long function or to follow DRY pattern).